src/reconcile/rec_visibility.c

```diff
@@ -292,6 +302,37 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         if (WT_TXNID_LT(max_txn, txnid))
             max_txn = txnid;
 
+        /*
+         * If we're seeing out-of-order updates, we need to pin these updates into the cache.
+         *
+         * We're also clearing the selected update since nothing newer than this out-of-order update
+         * is allowed to be written to the disk. If we find something older in the update list that
+         * can be written to the disk, that's ok and we should go ahead and select it.
+         *
+         * If the previous update is globally visible, we don't need to care about out-of-order ids
+         * or timestamps since the offending update will usually be removed by an obsolete check
+         * further down the line. If there is a non-globally visible update further in the chain,
+         * the obsolete check won't remove these updates. However, that non-globally visible update
+         * must be out-of-order by definition and will therefore trigger this check and pin the
+         * list.
+         */
+        if (!F_ISSET(r, WT_REC_IN_MEMORY) && prev_upd != NULL && prev_upd->txnid != WT_TXN_ABORTED)
```

Comment:
prev_upd->txnid != WT_TXN_ABORTED Is this check now redundant?

Reply:
223a182